source/dynamical_systems/README.md

```diff
@@ -1,2 +1,115 @@
+to the distance of the current state from the attractor.
+
+It is currently implemented for the `CartesianState` and `JointState` types, and
+can be constructed as follows:
+
+```c++
+StateRepresentation::CartesianState csA("A");
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(csA);
+
+StateRepresentation::JointState jsB("B");
+DynamicalSystem::Linear<StateRepresentation::JointState> linear(jsB);
+```
+
+### Configuring the Linear DS
+
+To change the strength of the attractor, a gain can be passed as the second argument during construction or
+passed to the `set_gain()` member function. The gain defines the proportionality between
+a distance unit and a velocity unit, and is internally stored as a square matrix with a size corresponding
+to the degrees of freedom in the state representation. For example, the `CartesianState` has six degrees of freedom
+(XYZ in linear and angular space), while the `JointState` would have as many degrees of freedom as joints.
+The gain can be defined as a matrix directly, as a diagonal vector of the appropriate
+length, or as a scalar (which sets the value along the diagonal elements of the matrix).
+```c++
+// set a gain (scalar, vector or matrix during construction)
+double gain = 10;
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(sA, gain);
+
+// or set / update the gain for the created object
+std::vector<double> gains = {1, 2, 3, 4, 5, 6};
+linear.set_gain(gains)
+```
+
+### Evaluating the Linear DS
+
+To get the twist, simply call the `evaluate` method.
+
+```c++
+StateRepresentation::CartesianState csA("A"), csB("B");
+DynamicalSystem::Linear<StateRepresentation::CartesianState> linear(csA);
+
+// note: the return type of evaluate is a CartesianState, but
+// it can be directly assigned to a CartesianTwist because the =operator
+// has been defined for that purpose
+StateRepresentation::CartesianState twist = linear.evaluate(csB);
+```
+
+### Reference frames
+
+The following section applies for the `CartesianState` type.
+ 
+On construction, the attractor is given the same name and reference frame as
+ the provided state. The evaluate function will always output the state
+ in that reference frame. The state that is evaluated must be connected to the
+ reference frame of the DS, by either having the same name or the same reference frame.
+```c++
+StateRepresentation::CartesianState BinA("B", "A");
+StateRepresentation::CartesianState CinA("C", "A");
+StateRepresentation::CartesianTwist twist;
+
+// initialise the linearDS with attractor at B in reference frame A
+DynamicalSystems::Linear<StateRepresentation::CartesianState> linearDS(BinA);
+
+// evaluate the twist for state C in reference frame A
+twist = linearDS.evaluate(CinA); // valid
+
+// evaluate the twist for the reverse expression: state A in reference frame C
+auto AinC = CinA.inverse();
+twist = linearDS.evaluate(AinC); // valid, same output as evaluate(CinA)
```

Comment:
I don't understand. Why do these two twists give the same output? And why do we want that to happen?